projects/opendr_ws_2/src/data_generation/data_generation/synthetic_facial_generation.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3.6
+# Copyright 2020-2022 OpenDR European Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cv2
+import torch
+import rclpy
+from ros2_bridge.bridge import ROS2Bridge
+import numpy as np
+from sensor_msgs.msg import Image as ROS_Image
+from SyntheticDataGeneration import MultiviewDataGeneration
+import os
+import argparse
+from opendr.engine.data import Image
+from algorithm.DDFA.utils.ddfa import str2bool
+
+
+class Synthetic_Data_Generation:
```

Comment:
In ROS2 the node class should inherit from `Node`, imported as `from rclpy.node import Node`. 

Also, for consistency reasons with the rest of the nodes and toolkit classes: [`Class names should normally use the CapWords convention.`](https://peps.python.org/pep-0008/#class-names)

```suggestion
class SyntheticDataGeneration(Node):
```